esi/src/parse.rs

```diff
@@ -1,85 +1,118 @@
 use crate::{ExecutionError, Result};
 use log::debug;
-use quick_xml::events::BytesStart;
+use quick_xml::events::{BytesStart, Event as XmlEvent};
+use quick_xml::name::QName;
 use quick_xml::Reader;
 use std::io::BufRead;
+use std::ops::Deref;
+
+// State carrier of Try branch
+#[derive(Debug, PartialEq)]
+enum TryTagArms {
+    Try,
+    Attempt,
+    Except,
+}
 
 /// Representation of an ESI tag from a source response.
 #[derive(Debug)]
-pub enum Tag {
+pub struct Include {
+    pub src: String,
+    pub alt: Option<String>,
+    pub continue_on_error: bool,
+}
+
+#[derive(Debug)]
+pub enum Tag<'a> {
     Include {
         src: String,
         alt: Option<String>,
         continue_on_error: bool,
     },
+    Try {
+        attempt_events: Vec<Event<'a>>,
+        except_events: Vec<Event<'a>>,
+    },
 }
 
 /// Representation of either XML data or a parsed ESI tag.
 #[derive(Debug)]
 #[allow(clippy::upper_case_acronyms)]
 pub enum Event<'e> {
-    XML(quick_xml::events::Event<'e>),
-    ESI(Tag),
+    XML(XmlEvent<'e>),
+    ESI(Tag<'e>),
 }
 
-/// Parses the ESI document from the given `reader` and calls the `callback` closure upon each successfully parsed ESI tag.
-pub fn parse_tags<'a, R>(
-    namespace: &str,
+// #[derive(Debug)]
+struct EsiTags {
```

Comment:
I think `TagNames` is a more accurate representation of this struct.
```suggestion
struct TagNames {
```

Reply:
🤔 well... kind of disagree, but since the ESI is the name of the crate itself, okay. I'll update this